src/Logary/Internals/SpanModule.fs

```diff
@@ -3,83 +3,74 @@ namespace Logary
 open System.Threading
 open System
 open NodaTime
-open FSharp.NativeInterop
-
-
-/// describe the time scope info about a span, will be sent as a message's context data
-[<Struct>]
-type SpanInfo =
-  {
-    id : string
-    beginAt: int64 // number of ticks since the Unix epoch. Negative values represent instants before the Unix epoch. (from NodaTime)
-    endAt: int64 // number of ticks since the Unix epoch. Negative values represent instants before the Unix epoch. (from NodaTime)
-    duration: int64 // total number of ticks in the duration as a 64-bit integer. (from NodaTime)
-  }
 
 [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
 module SpanBuilder =
 
-  type private T =
+  /// describe the time scope info about a span, will be sent as a message's context data
+  [<Struct>]
+  type internal SpanLog =
     {
+      traceId: string
+      spanId: string
       parentSpanId: string
-      messageFac: LogLevel -> Message
-      clock: IClock
-      localRootPrefix: string
+      beginAt: int64 // number of ticks since the Unix epoch. Negative values represent instants before the Unix epoch. (from NodaTime)
```

Comment:
We use NanosSinceEpoch elsewhere; let's not conflate them, please. Ticks <> nanos, but a .Net-specific thing that is 10 000 * something I can never remember (millis?).